### ISSUE TITLE
Ability to manually update reports

### ISSUE THREAD

**Issue by shrayshray** [datamade/la-metro-councilmatic#658]
Create a way to manually update/sync reports by report ID or Matter ID. Metro would like to be able to enter ID of files we know need to be updated on the site because we were alerted a change was made in Legistar. This feature could live either on the dashboard, or in the admin interface. And wherever this feature lives, we would like there to also be a log/list of items manually requested updates and completed requests since last full scrape.

**Comment by hancush** [comment]
When updates are made day/s before meeting, board agendas are not updated on the site until the next day. Is updated attachment captured by condition to scrape board reports included on future agendas? We fixed a bug in this [last summer](https://github.com/datamade/django-councilmatic/pull/268), but that's obviously before this issue occurred. @shrayshray received a report of this occurring as recently as Feb. / March of this year.

**Comment by camachoo** [comment]
I just updated a report during the normal course of business and thought it might be a good candidate for this issue,  I updated the related reports field. Let's see if this updates.

[https://boardagendas.metro.net/board-report/2020-0582/](url)

Next, we should also look at updating an attachment to see if that triggers an updated date for the report.

**Comment by hancush** [comment]
So, I was mistaken: We implemented logic to update the document cache for board reports on upcoming agendas, but we never updated the scrape logic. I've drafted [a change to the scraper to scrape bills on upcoming agendas](https://github.com/opencivicdata/scrapers-us-municipal/pull/344), will be in touch re: testing next week.

**Comment by hancush** [comment]
@shrayshray @camachoo I've deployed the change in scraping behavior to the staging dashboard, which will populate the database on the staging board agendas site. Could we do a round of test data entry next week? Per my understanding, that would look like:

- Create a test board report
- Create a test event scheduled for a future date, then add the test board report to its agenda
- Wait an hour for the board report to fall out of the update window, then make a change to the board report that does not toggle its updated at time stamp. The example you cited could provide a good test case. Was it that the board report attachments were updated?
- Confirm that the test board report is captured in the next windowed scrape and that the change appears on the site

Let me know if you have any questions, or otherwise when would be convenient for you to coordinate on the test. (Meanwhile, I'm on support tonight and wishing you both happy agenda posting and a nice weekend!)

**Comment by shrayshray** [comment]
@hancush yes, we can test this next week, likely Tuesday. Looking forward to an uneventful agenda posting :) and hope you have a nice weekend!

**Comment by shrayshray** [comment]
@hancush after publishing a test meeting (8/31/21 Regular Board) with test report (2021-0382), and waiting an hour for it to fall out of the update window, I tested the following scenarios:
1. Relating another report to the test report. This did not modify any field in the log, but the change showed on the site anyway!
2. Adding an attachment. This changed the field TBLATTACHMENT. It showed on the site as expected. 
3. Replacing the attachment and changing the attachment name. Both modified changes modified the field TBLATTACHMENT. This change showed as expected.
4. Removing the related report. This change showed in the log as modifying the field TBLMATTERRELATION - odd that adding a related report didn't log a change to this field.

All of these changes to re report showed on the staging site - and also on the live site! So I'm wondering if maybe I missed a test case? I've since changed the meeting to "Draft" and the report to "TO BE REMOVED" but can bring them back if there are other things to test.

**Comment by hancush** [comment]
@shrayshray I underestimated the window! We use a window (which represents a proportion of a day) of 0.05. That figures to 0.05 * 24 = 1.2 hours, or 72 minutes. Did you make these changes fairly quickly at the hour? If so, it makes sense that they showed up on the live site, because they still fell within the window. What do you think about repeating the test, but this time waiting _two_ hours between report creation and updates? Sorry for the snafu!

**Comment by shrayshray** [comment]
@hancush I did make the changes fairly quickly - exported the agenda at 3:45pm and made the first update at 4:51pm.
I repeated the test today, exporting the agenda at 11:25am but waiting until 2:15pm to make changes. I tested the same 4 scenarios as [last time](https://github.com/datamade/la-metro-councilmatic/issues/658#issuecomment-850594316), and got the same result - all showed up within minutes!

**Comment by hancush** [comment]
@camachoo will review the Granicus changelogs and report any potentially related updates, e.g., changes in last updated timestamp behavior. @shrayshray will look for report of board report not updating from earlier this year. We'll use this info to determine next steps on the development side.
